pkg/utils/pod/scheduling.go

```diff
@@ -39,13 +39,15 @@ func IsActive(pod *corev1.Pod) bool {
 // - Is an active pod (isn't terminal or actively terminating) OR Is owned by a StatefulSet and Is Terminating
 // - Isn't owned by a DaemonSet
 // - Isn't a mirror pod (https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/)
+// - Does not have the "karpenter.sh/do-not-disrupt=true" annotation (https://karpenter.sh/docs/concepts/disruption/#pod-level-controls)
 func IsReschedulable(pod *corev1.Pod) bool {
 	// StatefulSet pods can be handled differently here because we know that StatefulSet pods MUST
 	// get deleted before new pods are re-created. This means that we can model terminating pods for StatefulSets
 	// differently for higher availability by considering terminating pods for scheduling
 	return (IsActive(pod) || (IsOwnedByStatefulSet(pod) && IsTerminating(pod))) &&
```

Comment:
What do you think about adding the PDB handling in here so that we have all of the logic for pods that we think are reschedulable and should be considered in one place?

Reply:
The reason I didn't do this is that this is used to compute reschedulable pods for candidate selection which is then used in a lot of places to determine things like empty nodes, etc. Felt like there could be unintentional side-effects since I don't fully understand those code paths yet hence trying to make this a targeted change. What do you think?